digitalocean_plugin/tests/test_security.py

```diff
@@ -0,0 +1,278 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+import os
+import testtools
+import random
+import responses
+
+from cloudify.mocks import MockCloudifyContext
+from cloudify.exceptions import NonRecoverableError
+
+from digitalocean_plugin.security import DigitalOceanSecurity
+
+
+class TestSecurity(testtools.TestCase):
+
+    test_pubkey_filename = 'testkey.pub'
+    badkey_path = 'xyzpdq.pub'
+    test_random = random
+
+    test_instance = DigitalOceanSecurity()
+
+    @staticmethod
+    def mock_ctx(test_name):
+
+        test_node_id = test_name
+        test_properties = {}
+
+        ctx = MockCloudifyContext(node_id=test_node_id,
+                                  properties=test_properties)
+
+        return ctx
+
+    @staticmethod
+    def make_url(end_of_url):
+        return "https://api.digitalocean.com/v2/%s" % end_of_url
+
+    @staticmethod
+    def load_response(fixture_filename):
+
+        cwd = os.path.dirname(__file__)
+        fix_path = "%s/fixtures/" % cwd
```

Comment:
You might consider creating temp files that will be easily removed like this: https://github.com/cloudify-cosmo/cloudify-aws-plugin/blob/master/ec2/tests/test_ec2_utils.py#L79
Then you can add removal of that file to the cleanup class: https://github.com/cloudify-cosmo/cloudify-aws-plugin/blob/master/ec2/tests/test_ec2_utils.py#L82